src/main/java/net/sf/jabref/logic/search/rules/GrammarBasedSearchRule.java

```diff
@@ -146,10 +148,28 @@ public boolean compare(BibEntry entry) {
                 return matchFieldValue(entry.getType());
             }
 
+            // special case for searching a single keyword
+            if (fieldPattern.matcher("keyword").matches()) {
+                if (entry.hasField(FieldName.KEYWORDS)) {
+                    List<String> keywords = new ArrayList<>(entry.getKeywords());
```

Comment:
The getFieldOptional is no more. ;-)

You mean:
`return entry.getKeywords().stream().anyMatch(this::matchFieldValue);`
?

Could work. In the most recent version I swapped order of hasField and matcher as I expected the first one to be faster. Still, I think getKeywords() returns an empty list if the field is not set, so doesn't really need it.